### PR TITLE
[MM-28262] e2e/messaging: add T2167

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Add MM-T2167 - Pin a post, view pinned posts.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28262

### CODE REVIEW

File: e2e/cypress/integration/messaging/pinned_posts_spec.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            // # On a message in center channel, click then pin the post to the channel
+            cy.getPostMenu(postId, 'Pin to Channel').click();
+
+            // * Find the 'Pinned' span in the post pre-header to verify that the post was actually pinned
+            cy.get(`#post_${postId}`).findByText('Pinned').should('exist');
+
+            // # Click pin icon next to search box
+            cy.get('#channelHeaderPinButton').should('exist').click();
+
+            // * RHS title displays as "Pinned Posts in [channel name]"
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/pinned_posts_spec.js:44]: I've updated this in test case, read as: `RHS title displays as "Pinned Posts" and "[channel name]"`

File: e2e/cypress/integration/messaging/pinned_posts_spec.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        });
+    });
+
+    it('MM-T2167 Pin a post, view pinned posts', () => {
+        // # Login as the other user
+        cy.apiLogin(testUser);
+
+        // # Visit a test channel and post a message
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+        cy.postMessage('This a post that is going to be pinned.');
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/pinned_posts_spec.js:32]:
```suggestion
        cy.postMessage('This is a post that is going to be pinned.');
```

**Review by saturninoabril — COMMENTED** [review]
Thanks @isacikgoz! Tested and passed. Looks good to me but left minor comments.

@lindy65 For the second bullet, which is:
`Message has Pinned badge in center and RHS.`

Should that be `... in center but not in RHS`? I don't think we're showing badge on "Pinned Posts" RHS.

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

File: e2e/cypress/integration/messaging/pinned_posts_spec.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        // # Visit a test channel and post a message
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+        cy.postMessage('This is a post that is going to be pinned.');
+
+        cy.getLastPostId().then((postId) => {
+            // # On a message in center channel, click then pin the post to the channel
+            cy.getPostMenu(postId, 'Pin to Channel').click();
+
+            // * Find the 'Pinned' span in the post pre-header to verify that the post was actually pinned
+            cy.get(`#post_${postId}`).findByText('Pinned').should('exist');
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/pinned_posts_spec.js:38]: I'm all good, except for this one and will probably update the test case to something like,
`Message has Pinned badge in center but not in "Pinned Posts" RHS.`

With that, could you please add verification that the badge does not exist in RHS?

File: e2e/cypress/integration/messaging/pinned_posts_spec.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+            // # Click pin icon next to search box
+            cy.get('#channelHeaderPinButton').should('exist').click();
+
+            // * RHS title displays as "Pinned Posts in [channel name]"
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/pinned_posts_spec.js:43]:
```suggestion
            // * RHS title displays as "Pinned Posts" and "[channel name]"
```

**Review by saturninoabril — APPROVED** [review]
@isacikgoz Thanks for updating, LGTM!

**Review by lindy65 — APPROVED** [review]
Thanks @isacikgoz - looks good to me 👍 

Yes @saturninoabril - "pinned" badge is not displayed in RHS. I've updated the test case.

**Pull request merged**